### ISSUE TITLE
Using the app BankID to scan QR codes on OnePlus6 Enchilada crashes the phones and goes in to "QUALCOMM CrashDump"

### ISSUE THREAD

**Issue by Arash-SJPI22** [Divested-Mobile/DivestOS-Build#192]
I have an OnePlus 6 with enchilada version 20-20230213-dos-enchilada on it with the app BankID version 7.31.0.126.
This problem appeard for the first time 24/2 -23 when I opened my BankID to scan a QR code for verification. As soon as I pressed the button on the app to scan QR codes it went to the next step but the camera was showing a blackscreen and the app froze and a couple of seconds later it went in to "QUALCOMM CrashDump".
I was able to restart the phone without any problems but when I opened the BankID app again my ID's where blocked and unusable. I had to go to my bank to get a new ID for the app. And when I tried the same thing to day the same thing happened and now my new ID is also blocked.
I have no other problems with the phone or camera. I can scan QR codes with other apps without crashing. 
And now when my ID's are blocked it dosen't crash when I try to scan QR codes.

The BankID app have always complained that it wont run without Google Play services on start but have always worked without problems.
I don't think I have used the QR scan function on the BankID app after I upgraded to the last version released on 13th february. So maybe the problem started after that version.

**Comment by SkewedZeppelin** [comment]
bugs like this with latent corruption after camera use aren't expected on end of life devices like this

I've seen before that enchilada/fajita already has memory corruption issues with QR handling, but it never crashes the kernel: https://issuetracker.google.com/issues/232569772

**Comment by Arash-SJPI22** [comment]
Thank you for the link, I saw that it was marked fixed in june 2022 in that thread. Dose this mean there wont be an attempt to fix it? 
Can I help out by providing logs or anything else?

**Comment by SkewedZeppelin** [comment]
is there a short error shown on the crashdump screen?

also ftr the quick way to get out of crashdump: hold volume up + power to exit

**Comment by Arash-SJPI22** [comment]
Nothing just "QUALCOMM CrashDump Mode" on the top of the screen.
Yes, tnx I was able to find that shortcut after some time.

**Comment by danielk43** [comment]
Could [this](https://github.com/LineageOS/android_vendor_lineage/blob/lineage-20.0/overlay/common/frameworks/base/core/res/res/values/config.xml#L176) possibly be an issue?

i.e., `org.lineageos.aperture` should be replaced with `app.grapheneos.camera`, maybe

**Comment by SkewedZeppelin** [comment]
@danielk43 nah, a ton of device camera drivers are laden with memory corruption issues
and 20.0 doesn't use GrapheneOS camera

**Comment by Arash-SJPI22** [comment]
Here comes part of the kernel log.
[partofKMSG.txt](https://github.com/Divested-Mobile/DivestOS-Build/files/10875004/partofKMSG.txt)
Let me know if more is needed.

**Comment by SkewedZeppelin** [comment]
To confirm that crash log happened after opening its QR scanner?

The error is an out of bounds operating from dm-verity, not camera related.

I wonder if you did it again you'd get a different error or the same?
If it was different it would confirm camera related corruption, if it wasn't that'd be a bit more of an issue.

**Comment by Arash-SJPI22** [comment]
Yes, after it opens the camera to scan QR codes.
I am not too keen on the idea to try it again, since I was just at the bank and got a new ID on it. And every time it crashes it makes my ID's invalid. If you really need another crash log I can do it, but rather not.
I'll play around and see if I can get it to crash with another QR scanning or camera app.

**Comment by Arash-SJPI22** [comment]
I got another "QUALCOMM CrashDump Mode", but this time it was not from the BankID app. I was just browsing using Mull and then it crashed. Here is part of the KMSG.txt from "edl memorydump". (This time it didn't use the app).
[partof2ndKMSG.txt](https://github.com/Divested-Mobile/DivestOS-Build/files/11070192/partof2ndKMSG.txt)

And right now while writing this and doing nothing on the phone it just crashed again to CrachDump Mode.., I had just restarted it, or rather it restarted it self after "edl memorydump" was done and I had entered pin and so on and it had turned of its screen. I had not opened any apps and was not using it, but it was connected to my computer by USB.
So here is another part of the KMESG from the latest crash...:(
[partof3rdKMSG.txt](https://github.com/Divested-Mobile/DivestOS-Build/files/11070212/partof3rdKMSG.txt)

**Comment by SkewedZeppelin** [comment]
Please install the latest update and see if you still have issues on it

**Comment by Arash-SJPI22** [comment]
Just updated and it crashed again while using Mull and just browsing (a different site then last time).
[partof4thKMSG.txt](https://github.com/Divested-Mobile/DivestOS-Build/files/11071010/partof4thKMSG.txt)

Although I saw there is a new version of Mull, so I'll update and see if it will fix it.

**Comment by SkewedZeppelin** [comment]
I may be completely wrong, but this is likely a hardware issue of your storage failing, as the errors posted are dm-verity data verification failures.

**Comment by Arash-SJPI22** [comment]
Uhoh ... I hope not, but still it's strange that it only crashes while using this app and only while scanning QR-codes. (Except for the two other crashes while using Mull, but those were fixed by the update and haven't happened since). It has nothing to do with it being a security app? (just a shot in the dark)

**Comment by SkewedZeppelin** [comment]
are you still encountering this? change OS? change device?

**Comment by Arash-SJPI22** [comment]
Yes, but it has gotten worse and crashes more often, seemingly at random. 
Still same device, same OS but updated to the latest version.
Probably some hardware issue, as you guessed previously.
